### PR TITLE
Added cache usage for RBAC DbManager

### PR DESCRIPTION
**Pull request by manchenkoff**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | #15376

### CODE REVIEW

File: framework/rbac/DbManager.php [hunk 1]
```diff
@@ -471,6 +475,14 @@ public function getRolesByUser($userId)
             return [];
         }
 
+        if ($this->cache instanceof CacheInterface) {
```
Comment (bizley) [inline framework/rbac/DbManager.php:478]: init() takes care of that check so we only need to see if it's not null

File: framework/rbac/DbManager.php [hunk 2]
```diff
@@ -1055,4 +1072,9 @@ protected function isEmptyUserId($userId)
     {
         return !isset($userId) || $userId === '';
     }
+
+    private function getUserRolesCacheKey()
+    {
+        return $this->cacheKey + '.' + $this->rolesCacheSuffix;
```
Comment (bizley) [inline framework/rbac/DbManager.php:1078]:
```suggestion
        return $this->cacheKey . $this->rolesCacheSuffix;
```

File: framework/rbac/DbManager.php [hunk 3]
```diff
@@ -482,6 +494,10 @@ public function getRolesByUser($userId)
             $roles[$row['name']] = $this->populateItem($row);
         }
 
+        if ($this->cache instanceof CacheInterface) {
```
Comment (bizley) [inline framework/rbac/DbManager.php:497]: same comment as above here

File: framework/rbac/DbManager.php [hunk 4]
```diff
@@ -105,6 +105,10 @@ class DbManager extends BaseManager
      * @since `protected` since 2.0.38
      */
     protected $checkAccessAssignments = [];
+    /**
+     * @var string the key used to store user RBAC roles in cache
+     */
+    protected $rolesCacheSuffix = '.roles';
```
Comment (bizley) [inline framework/rbac/DbManager.php:111]: Dot is not needed here. Also it suppose to be settable so public.

**Review by machour — APPROVED** [review]

**Review by bizley — APPROVED** [review]

**Review by Arhell — APPROVED** [review]

File: framework/rbac/DbManager.php [hunk 5]
```diff
@@ -87,6 +87,10 @@ class DbManager extends BaseManager
      * @since 2.0.3
      */
     public $cacheKey = 'rbac';
+    /**
+     * @var string the key used to store user RBAC roles in cache
```
Comment (samdark) [inline framework/rbac/DbManager.php:91]:
```suggestion
     * @var string the key used to store user RBAC roles in cache
     * @since 2.0.48
```

**Comment by samdark** [comment]
👍

**Pull request merged**